src/com/aca/naresilanyan/streamAPI/streamAPI.java

```diff
@@ -0,0 +1,224 @@
+package com.aca.naresilanyan.streamAPI;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+public class streamAPI {
+    public static void main(String[] args) {
+//   1.     Strings
+        String stringTest = string(Arrays.asList(1,2,3,4));
+        System.out.println(stringTest);
+
+
+//   3.   Get the highest populated city of each country
+
+
+        Country country = new Country("us", "United States", "North America", 328,9833520, 20.84,"Washington");
+        Country country2 = new Country("de", "Germany", "Europe", 83,357022, 4.5,"Berlin");
+
+
+        country.setCities(Arrays.asList(
+                new City(2,"Chicago", "us",2695598),
+                new City(10,"New York", "us",8175133),
+                new City(3,"Los Angeles","us", 379621)
+        ));
+
+        country2.setCities(Arrays.asList(
+                new City(4,"Humburg", "de",1787000),
+                new City(6,"Munich", "de",1450000),
+                new City(7,"Berlin","de", 3520000)
+
+        ));
+        List<Country> countries = Arrays.asList(country, country2);
+
+        Map<String, String> countryCityMap= new HashMap<>();
+
+
+        countries.stream().forEach(country1 -> {
+           String cityName = country1.getCities().stream().max((o1,o2)-> o1.getPopulation()-o2.getPopulation()).stream().findFirst().get().getName();
```

Comment:
why call stream() the second time ?

Reply:
I guess you had warrning on _**findFirst().get()**_ part, findFirst is Optional so please get from it some value after checking it is present or not